### DISCUSSION TITLE
Add support for new experimental boolean operation for vector images

### DISCUSSION THREAD

**Discussion by JonasMht** [godotengine/godot-proposals#4927] [Engine Core]
The new experimental live path effect called boolean operation is not supported for vector images in the GodotEngine.

As an example i use an image with a boolean operation of two overlayed triangles resulting in the following image :
![Down_Icon](https://user-images.githubusercontent.com/47430264/179950920-2d2e2dea-78b5-41fc-bbac-88112beaa544.svg)

In the GodotEngine the boolean operation is not applied and we can see both triangles:
![Capture d’écran du 2022-07-20 11-34-27](https://user-images.githubusercontent.com/47430264/179949810-50a04e02-5206-45d6-8e25-fe077c89923e.png)

For more informations about these path functions:
[https://wiki.inkscape.org/wiki/index.php/Release_notes/1.0#Boolean_Operation_LPE](https://wiki.inkscape.org/wiki/index.php/Release_notes/1.0#Boolean_Operation_LPE)

Wouldn't it be great if this feature could be added to the svg importer ?

**Comment by Calinou** [comment]
Godot 3.x uses the [NanoSVG](https://github.com/memononen/nanosvg) library, which only supports a subset of the SVG 1.0 specification. Godot 4.0.alpha uses [ThorVG](https://www.thorvg.org/), which is more capable and robust, but still aims to support SVG 1.0 only.

Most importantly, **neither NanoSVG nor ThorVG support any Inkscape-specific functionality** including Live Path Effects. To get your SVG to render correctly in Godot, you will have to find a way to "bake" your SVG to use standard SVG 1.0 features only. Alternatively, you can export the SVG to a PNG image to be imported by Godot.

**Reply by JonasMht** [reply]
Very interesting to know. Thank you for your answer!
